pkg/lib/get-timesync-backend.py

```diff
@@ -0,0 +1,44 @@
+import os
+import subprocess
+
+# get-timesync-backend - determine which NTP backend timedatectl will
+#                        likely enable and whether we support it
+
+roots = ["/etc", "/run", "/usr/local", "/usr/lib"]
+
+
+def gather_files(name, suffix):
```

Comment:
This doesn't do the same.  The original code would avoid duplicates and would sort differently.  For example, we want `/usr/lib/.../10-chrony.list` to come before `/etc/.../20-timesyncd.list`, but your code would put them the other way around.

Reply:
Ah, so your *intention* is that the returned list sorts by file name ("10-chrony"), not by file path. But your code sorts the list with whole paths alphabetically, i.e. `[files[f] for f in sorted(files.keys())]`, so any /etc/ would come before /usr.  My code skipped the sorting indeed.

So I suppose you need to give sorted() a sort function which compares basenames only. Then the glob version should work just fine as well.

Reply:
Also, this is a really weird way to sort a list -- `files.sort()` is easier, faster, and avoids the extra *three* copies.

Reply:
Oh, sorry -- I see, `files` is a dictionary indexed by file names. This is really not that obvious with all the single-letter variable names and the deep nesting. Please at least add a comment to it what it does and why.

But I still think this is unnecessarily convoluted. `sort()`in by basename works just fine:
```
>>> units = ['/usr/10foo', '/etc/20foo', '/usr/30foo']

>>> sorted(units)
['/etc/20foo', '/usr/10foo', '/usr/30foo']
# that's not intended

>>> sorted(units, key=os.path.basename)
['/usr/10foo', '/etc/20foo', '/usr/30foo']
# that works
```

Reply:
> This is still not good enough -- there may be (and often are) files with the same name in both /usr/ and /etc. These need to be sorted as /etc < /usr/local < /usr.

The original code does this.  Ir processes the roots in order and does not replace an entry in files if one already exists.  Thus if a given basename is found in both /etc and /usr, the one from /etc ends up in `files`.

Reply:
Ah, ok. This is a really cryptic piece of code :cry: Honestly  I hate this, but then again, I hate just about everything on this PR.. So I'm just wearing out on saying "no". It would just *really* be nice to add a comment here about what this does, and its semantics and output.

Reply:
Heh, at least give me credit for using a list comprehension. :-)

I have changed the variable names and added a comment.

Reply:
Sorry if that came across wrong -- I meant I really dislike that a frontend UI has to even do all of this nonsense in the first place. I'm also not a fan of the code complexity here. Anyway, apologies! Thanks for the comment.